code/modules/surgery/organs/autosurgeon.dm

```diff
@@ -103,9 +103,61 @@
 	uses = 1
 	starting_organ = /obj/item/organ/cyberimp/arm/syndie_mantis
 
-/obj/item/autosurgeon/organ/syndicate/syndie_mantis/l
+/obj/item/autosurgeon/organ/syndicate/syndie_mantis/AltClick(mob/living/user) //Basically this combines screwdriver with an insert_organ proc because I can't be assed
+	var/obj/item/I = storedorgan
+	var/atom/drop_loc = user.drop_location()
+	if(!storedorgan)
+		return
+	for(var/J in src)
+		var/atom/movable/AM = J
+		AM.forceMove(drop_loc)
+	if(!istype(I, /obj/item/organ/cyberimp/arm/syndie_mantis/l))
+		for(var/obj/item/organ/cyberimp/arm/syndie_mantis/R in drop_loc)
+			R.Destroy()
+		var/obj/item/organ/O = new /obj/item/organ/cyberimp/arm/syndie_mantis/l(src)
+		insert_organ(O)
+		to_chat(user, span_notice("You change the autosurgeon to target the left arm."))
+		return
+	if(istype(I, /obj/item/organ/cyberimp/arm/syndie_mantis/l))
+		for(var/obj/item/organ/cyberimp/arm/syndie_mantis/l/L in drop_loc)
+			L.Destroy()
+		var/obj/item/organ/O = new /obj/item/organ/cyberimp/arm/syndie_mantis(src)
+		insert_organ(O)
+		to_chat(user, span_notice("You change the autosurgeon to target the right arm."))
+		return
+
+/obj/item/autosurgeon/organ/syndicate/syndie_hammer
 	uses = 1
-	starting_organ = /obj/item/organ/cyberimp/arm/syndie_mantis/l
+	starting_organ = /obj/item/organ/cyberimp/arm/syndie_hammer
+
+/obj/item/autosurgeon/organ/syndicate/syndie_hammer/attack_self(mob/user) //Preternis-only implant (if you don't manually remove the implant)
+	if(!ispreternis(user))
+		to_chat(user, span_warning("The autosurgeon rejects your body!"))
+		return
+	..()
+
+/obj/item/autosurgeon/organ/syndicate/syndie_hammer/AltClick(mob/living/user) //See mantis code
+	var/obj/item/I = storedorgan
+	var/atom/drop_loc = user.drop_location()
+	if(!storedorgan)
+		return
+	for(var/J in src)
+		var/atom/movable/AM = J
+		AM.forceMove(drop_loc)
+	if(!istype(I, /obj/item/organ/cyberimp/arm/syndie_hammer/l))
+		for(var/obj/item/organ/cyberimp/arm/syndie_hammer/R in drop_loc)
+			R.Destroy()
+		var/obj/item/organ/O = new /obj/item/organ/cyberimp/arm/syndie_hammer/l(src)
+		insert_organ(O)
+		to_chat(user, span_notice("You change the autosurgeon to target the left arm."))
+		return
+	if(istype(I, /obj/item/organ/cyberimp/arm/syndie_hammer/l))
+		for(var/obj/item/organ/cyberimp/arm/syndie_hammer/l/L in drop_loc)
```

Comment:
why check drop loc and not just delete it while it's inside the implant instead

Reply:
Tried it, for some weird reason it wouldn't properly initialize a new one inside of it if it were deleted while inside the object.